tracetools_trace/tracetools_trace/tools/lttng_impl.py

```diff
@@ -342,12 +342,14 @@ def setup(
             # Per-user buffer
             buffer_type=lttngpy.LTTNG_BUFFER_PER_UID,
             channel_name=channel_name,
-            # Discard, do not overwrite
-            overwrite=0,
-            # We use 2 sub-buffers because the number of sub-buffers is pointless in discard mode,
-            # and switching between sub-buffers introduces noticeable CPU overhead
+            # Overwrite if snapshot mode, otherwise discard
+            overwrite=snapshot_mode,
+            # We use 2 sub-buffers in normal mode because the number of sub-buffers is pointless in
+            # discard mode, and switching between sub-buffers introduces noticeable CPU overhead.
+            # In snapshot mode, we use 4 sub-buffers to lose less data when sub-buffers are over-
+            # written, because when all sub-buffers are full the oldest one is discarded entirely.
```

Comment:
Good explanation! I would also add a link to this relevant section of the docs: https://lttng.org/docs/v2.13/#doc-channel-subbuf-size-vs-subbuf-count